### PR TITLE
feat(cli): Added reload-include and reload-exclude in litestar cli

### LINKED ISSUE #2875: Enhancement: Add ``reload-include`` from `uvicorn` CLI into `litestar` CLI
Reported by cj in Discord: <#1183485070451941559>:

Thank you both 🙂 That would be awesome if a `--reload-include` proxy was added to Litestar

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2875">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2875/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by ashm-tech**
Closes #2875

### CODE REVIEW

File: litestar/cli/_utils.py [hunk 1]
```diff
@@ -134,6 +136,8 @@ def from_env(cls, app_path: str | None, app_dir: Path | None = None) -> Litestar
         uds = getenv("LITESTAR_UNIX_DOMAIN_SOCKET")
         fd = getenv("LITESTAR_FILE_DESCRIPTOR")
         reload_dirs = tuple(s.strip() for s in getenv("LITESTAR_RELOAD_DIRS", "").split(",") if s) or None
+        reload_includes = [s.strip() for s in getenv("LITESTAR_RELOAD_INCLUDES", "").split(",") if s] or None
+        reload_excludes = [s.strip() for s in getenv("LITESTAR_RELOAD_INCLUDES", "").split(",") if s] or None
```
Comment (Alc-Alc) [inline litestar/cli/_utils.py:140]: is this supposed to be `LITESTAR_RELOAD_INCLUDES` for exclude?

Comment (ashm-tech) [reply litestar/cli/_utils.py:140]: Thanks for the review, it just stayed the same when copying, I'll change it now!)

And about the tests, I'll think about it

Comment (Alc-Alc) [reply litestar/cli/_utils.py]: You would also have to add these in https://github.com/litestar-org/litestar/blob/ca3659a3db5d9197b3341a38f54b7e0420f07424/docs/usage/cli.rst?plain=1#L127-L130 such that https://docs.litestar.dev/latest/usage/cli.html#cli-run-options shows the new env vars you have added. I am assuming others are ok with these names.

Comment (ashm-tech) [reply litestar/cli/_utils.py]: > You would also have to add these in
> 
> https://github.com/litestar-org/litestar/blob/ca3659a3db5d9197b3341a38f54b7e0420f07424/docs/usage/cli.rst?plain=1#L127-L130
> 
> such that https://docs.litestar.dev/latest/usage/cli.html#cli-run-options shows the new env vars you have added. I am assuming others are ok with these names.

I have updated the documentation, can you check and give advice if there is anything else you need to add besides tests? I will add tests soon!)

File: docs/usage/cli.rst [hunk 2]
```diff
@@ -143,6 +147,36 @@ To set multiple directories via an environment variable, use a comma-separated l
 
    LITESTAR_RELOAD_DIRS=.,../other-library/src
 
+--reload-include
+++++++++++++
```
Comment (JacobCoffee) [inline docs/usage/cli.rst:151]:
```suggestion
--reload-include
++++++++++++++++
```

File: docs/usage/cli.rst [hunk 3]
```diff
@@ -143,6 +147,36 @@ To set multiple directories via an environment variable, use a comma-separated l
...
+To set multiple directories via an environment variable, use a comma-separated list:
+
+.. code-block:: shell
+
+   LITESTAR_RELOAD_INCLUDE=.,../other-library/src
+
+--reload-exclude
+++++++++++++
+
+Set glob patterns to exclude while watching for files. Includes '.*, .py[cod], .sw.*, ~*' by default; these defaults can be overridden with `--reload-include`. This option has no effect unless watchfiles is installed. You can specify multiple directories by passing the flag multiple times:
```
Comment (JacobCoffee) [inline docs/usage/cli.rst:168]:
```suggestion
Set glob patterns to exclude while watching for files. Includes ``.*``, ``.py[cod]``, ``.sw.*``, ``~*`` by default; these defaults can be overridden with ``--reload-include``. This option has no effect unless `watchfiles <https://pypi.org/project/watchfiles/>`_ is installed. You can specify multiple directories by passing the flag multiple times:
```

**Review by JacobCoffee — CHANGES_REQUESTED** [review]

**Comment by JacobCoffee** [comment]
Thank you!
Could you add the matching tests in (i think) https://github.com/litestar-org/litestar/blob/main/tests/unit/test_cli/test_core_commands.py?

**Comment by ashm-tech** [comment]
> Thank you!
> Could you add the matching tests in (i think) https://github.com/litestar-org/litestar/blob/main/tests/unit/test_cli/test_core_commands.py?

Thank you too!) I will send a PR tomorrow with tests for new teams

**Comment by provinzkraut** [comment]
@ashm-tech Please rebase from `develop` and change the target to `develop`; This is a new feature and will go into the next minor release, which is being developed on that branch.

**Pull request closed without merging**